course_discovery/apps/api/v1/views/course_runs.py

```
@@ -244,6 +301,10 @@ def update(self, request, **kwargs):
         serializer = self.get_serializer(course_run, data=request.data, partial=partial)
         serializer.is_valid(raise_exception=True)
 
+        # First update the Seat(s) if the course run isn't already published
+        if course_run.status != CourseRunStatus.Published:
+            self.update_seats(mode, price, course_run)
```

Do we want to do anything if the price & mode don't match current values, but we're already published? Current code looks like it will silently ignore that request, which might be misleading. Maybe it doesn't super matter, if the UI doesn't let us get in that situation.

---

I was thinking no because right now, we send a patch request for every course run on every save and we don't want to update the Seat prices because of a change in the entitlement (which is where we are currently getting our mode/price from).

I wrote this code thinking that once a course run is published, we shouldn't really be changing the Seat prices for it, but we can still leave the ability to change the entitlement cost and have that apply to future runs without changing the older ones.

---

@transaction.atomic
    def _update_seats_from_entitlement(self, course, entitlement, changed_by):
        for run in self._get_active_course_runs(course):
            run.seats.all().delete()
            seat = Seat(
                type=CourseEntitlement.MODE_TO_SEAT_TYPE_MAPPING[entitlement.mode],
                price=entitlement.price,
                currency=entitlement.currency
            )

            # Use the SeatForm here to not duplicate logic for creating seats
            seat_form = SeatForm(instance=seat)
            seat_form.save(commit=True, course_run=run, changed_by=changed_by)

---

Based on this function in current publisher, we should continue to update seats even after publishing. I will update.